Sources/TreeConstructor/TreeSink.swift

```diff
@@ -0,0 +1,13 @@
+public enum QuirksMode: BitwiseCopyable, Equatable, Hashable, Sendable {
+    case quirks
+    case limited
+    case no
+}
+
+public protocol TreeSink<Handle>: ~Copyable {
+    associatedtype Handle
+
+    var document: Handle { get }
+    func parseError(msg: consuming String)
```

Comment:
Which should I use, `String` or `Str`?

Reply:
I've decided to use a dedicated type instead of a string.

https://github.com/kkebo/zyphy/compare/e51688c0545b29f37872983477e35d48651434a3..132dd5d2c526a94dda0baaf311cad8ba83cac73d